lib/OpenQA/Schema/Result/JobModules.pm

```diff
@@ -197,8 +197,8 @@ sub details {
     finally {
         close($fh);
     };
-    my $details = ref($ret) eq 'HASH' ? $ret : {results => $ret, execution_time => ''};
-    for my $img (@{$details->{results}}) {
+    my $results = ref($ret) eq 'HASH' ? $ret : {details => $ret};
+    for my $img (@{$results->{details}}) {
```

Comment:
Just a note, the "name swap" might've been nice to sum up in the PR description. I found this a bit confusing while attempting to trace what's going on here.

Reply:
Or add a comment in the code stating why this "name swap" is necessary. E.g.

```
# load result file as stored by os-autoinst (with hash-root)
# support also old format as stored by old openQA versions (with details as root)
```

Maybe it makes also sense to validate whether the details are actually an array-ref (and print a better warning in this case instead of letting Perl print the default warning).